src/main/java/net/pms/database/TableFilesStatus.java

```diff
@@ -95,18 +95,18 @@ public static void setFullyPlayed(final String fullPathToFile, final boolean isF
 			if (fileId == 0) {
 				query = "SELECT ID FROM FILES WHERE FILENAME = " + sqlQuote(fullPathToFile);
 				if (trace) {
-					LOGGER.trace("Searching for file with \"{}\" before update", query);
+					LOGGER.trace("Looking up file id for file \"{}\" with \"{}\" before update", fullPathToFile, query);
 				}
 
-				TABLE_LOCK.writeLock().lock();
+				TABLE_LOCK.readLock().lock();
 				try (Statement statement = connection.createStatement()) {
```

Comment:
You should only use a write lock when you're actually going to write. The difference between a read and a write lock is that a write lock is exclusive (only one can be held at any time) while read locks can be held in parallel with each other and will always be granted unless a write lock is held. Acquiring a write lock blocks until all other read- and write locks are released, and will block any new read- or write locks from being aquired while held. That means that as long as you hold a write lock, you have exclusive access to that table, so it should only be used when needed.